core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -493,4 +460,69 @@ protected int doHashCode() {
     public String getWriteableName() {
         return NAME;
     }
+
+    public static Rectangle parseBoundingBox(XContentParser parser) throws IOException, ElasticsearchParseException {
+        XContentParser.Token token = parser.currentToken();
+        if (token != XContentParser.Token.START_OBJECT) {
+            throw new ElasticsearchParseException("failed to parse bounding box. Expected start object but found [{}]", token);
+        }
+
+        double top = Double.NaN;
+        double bottom = Double.NaN;
+        double left = Double.NaN;
+        double right = Double.NaN;
+
+        String currentFieldName;
+        GeoPoint sparse = new GeoPoint();
+        EnvelopeBuilder envelope = null;
+
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+                token = parser.nextToken();
+                if (WKT_FIELD.match(currentFieldName)) {
+                    envelope = EnvelopeBuilder.class.cast(GeoWKTParser.parseExpectedType(parser, GeoShapeType.ENVELOPE));
```

Comment:
Can you change to the regular cast operator unless there is a specific reason for the Class.cast here? I think its more obvious whats happening and doesn't hide potential warnings etc... at compile time.